### PR TITLE
Feature/uppsf 2104 get to return latest version

### PR DESCRIPTION
**Pull request by StephanZahariev**
# Description

## What

GET to return the latest version

## Why

[UPPSF-2104](https://financialtimes.atlassian.net/browse/UPPSF-2104)

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [X] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: pkg/resources/patch.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ func PatchContent(mongo db.DB) func(w http.ResponseWriter, r *http.Request) {
 		tid := obtainTxID(r)
 		collectionID := mux.Vars(r)["collection"]
 		resourceID := mux.Vars(r)["resource"]
+		schemaVersion := r.Header.Get(SchemaVersionHeader)
+		schemaRevision := r.Header.Get(ContentRevisionHeader)
```
Comment (gotha) [inline pkg/resources/patch.go:30]: revision should always  be generated

File: pkg/resources/patch.go [hunk 2]
```diff
@@ -104,6 +108,7 @@ func PatchContent(mongo db.DB) func(w http.ResponseWriter, r *http.Request) {
 
 		w.Header().Add("Content-Type", contentTypeHeader)
 		w.Header().Add("Origin-System-Id", resource.OriginSystemID)
+		w.Header().Add(SchemaVersionHeader, schemaVersion)
```
Comment (gotha) [inline pkg/resources/patch.go:111]: I am not sure why are we returning the version header here. 
If anything, it makes more sense to return the revision

File: pkg/resources/utils.go [hunk 3]
```diff
@@ -31,19 +28,11 @@ func writeMessage(w http.ResponseWriter, msg string, status int) {
 func obtainTxID(req *http.Request) string {
```
Comment (gotha) [inline pkg/resources/utils.go:28]: I think here we should use transactionidutils.GetTransactionIDFromRequest and get rid of the txHeaderKey constant

File: pkg/resources/write.go [hunk 4]
```diff
@@ -40,21 +46,21 @@ func WriteContent(mongo db.DB) func(w http.ResponseWriter, r *http.Request) {
...
-			logger.WithMonitoringEvent("SaveToNative", tid, contentTypeHeader).WithUUID(resourceID).WithError(err).Error(msg)
+			logger.WithMonitoringEvent("SaveToNative", tid, contentType).WithUUID(resourceID).WithError(err).Error(msg)
 			http.Error(w, fmt.Sprintf("%s\n%v\n", msg, err), http.StatusInternalServerError)
 			return
 		}
 
-		logger.WithMonitoringEvent("SaveToNative", tid, contentTypeHeader).WithUUID(resourceID).Info(fmt.Sprintf("Successfully saved, collection=%s, origin-system-id=%s",
-			collectionID, originSystemIDHeader))
+		logger.WithMonitoringEvent("SaveToNative", tid, contentType).WithUUID(resourceID).Info(
+			fmt.Sprintf("Successfully saved, collection=%s, origin-system-id=%s schema-version=%s", collectionID, originSystemIDHeader, schemaVersion))
```
Comment (gotha) [inline pkg/resources/write.go:64]: just suggestion for improvement. instead of using fmt.Sprintf, we can use log.WithField("collection", "value"), etc. ..

**Review by gotha — CHANGES_REQUESTED** [review]

**Review by gotha — APPROVED** [review]

**Review by rnov — APPROVED** [review]

**Pull request merged**